packages/mobile/src/pincode/authentication.ts

```diff
@@ -1,94 +1,109 @@
+/**
+ * Logic and utilities for managing account secrets
+ * The pincode is a short numeric string the user is required to enter
+ * The pepper is a generated once per account and stored in the keychain/keystore
+ * The password is a combination of the two. It is used for unlocking the account in geth
+ */
+
 import { sha256 } from 'ethereumjs-util'
 import { asyncRandomBytes } from 'react-native-secure-randombytes'
+import { call, select } from 'redux-saga/effects'
 import { PincodeType } from 'src/account/reducer'
 import { pincodeTypeSelector } from 'src/account/selectors'
 import CeloAnalytics from 'src/analytics/CeloAnalytics'
 import { CustomEventNames } from 'src/analytics/constants'
+import { UNLOCK_DURATION } from 'src/geth/consts'
 import { navigate, navigateBack } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import {
   getCachedPassword,
   getCachedPasswordHash,
   getCachedPepper,
   getCachedPin,
+  setCachedPassword,
   setCachedPasswordHash,
   setCachedPepper,
+  setCachedPin,
 } from 'src/pincode/PasswordCache'
-import { ensureCorrectPin } from 'src/pincode/utils'
-import { store } from 'src/redux/store'
 import { retrieveStoredItem, storeItem } from 'src/storage/keychain'
 import Logger from 'src/utils/Logger'
-import { currentAccountSelector } from 'src/web3/selectors'
+import { getConnectedWalletAsync } from 'src/web3/contracts'
 
 const TAG = 'pincode/authentication'
+
 enum STORAGE_KEYS {
   PEPPER = 'PEPPER',
   PASSWORD_HASH = 'PASSWORD_HASH',
 }
+
 const PEPPER_LENGTH = 64
 export const PIN_LENGTH = 6
+// Pepper and pin not currently generalized to be per account
+// Using this value in the caches
+export const DEFAULT_CACHE_ACCOUNT = 'default'
+
+const PIN_BLACKLIST = [
```

Comment:
May be worth creating/throwing a custom error message for this to surface to the user. IMO may not be obvious why these are invalid.

Also would be nice to disable this with a dev flag because we've just blacklisted all of my go-tos ;)

Reply:
Yeah that's fair but I'm going to punt that to a follow-up PR. Juggling a few things atm and I think it's important to land this asap cause its risky.

Reply:
For the moment it shows 'Invalid Pin' to the user